### PR TITLE
docs: intro transaction skeleton

### LINKED ISSUE #697: Docs for the usage of `TransactionSkeleton`
Lately, we have received questions regarding the creation or updating of the `TransactionSkeleton`. As a result, it would be helpful to have documentation covering the following topics:

- fundamental understanding of CKB transaction
- basic knowledge of ImmutableJS
- explanation of the `cellProvider`, `fixedEntries`, and `signingEntries` fields
- instructions on how to create/update the `TransactionSkeleton`

### PR DESCRIPTION
**Pull request by homura**
# Description

Resolve #697

This PR introduces the usage of `TransactionSkeleton`

[Preview](https://github.com/ckb-js/lumos/blob/tx-skeleton-doc/website/docs/02-tx-skeleton.md)

## Type of change

Please delete options that are not relevant.

- [x] Documentation or Website

### CODE REVIEW

File: website/docs/02-tx-skeleton.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    // wrap the indexer as a CellProvider
+    indexer.collector({ ...query, type: "empty" }),
+})
+
+// update TransactionSkeleton
+txSkeleton = txSkeleton
+  .update("inputs", (inputs) => inputs.push(inputs0, inputs1))
+  .update("outputs", (outputs) => outputs.push(outputs0, outputs1))
+  .update("cellDeps", (cellDeps) => cellDeps.push(lockScriptDep, typeScriptDep0))
+  .upd.ate("witnesses", (witnesses) => witnessespush(aliceSignature))
```
Comment (Keith-CY) [inline website/docs/02-tx-skeleton.md]:
```suggestion
  .upd.update("witnesses", (witnesses) => witnesses.push(aliceSignature))
```

File: website/docs/02-tx-skeleton.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+// sign the tranasction
+const signatures = txSkeleton
+  .get("signingEntries")
+  .map(({ message }) => sign(message))
+  .toArray()
+
+// convert the TransactionSkeleton to RPC transaction
+const signedTransaction = sealTransaction(txSkeleton, signatures)
+
+// boardcast the transaction
```
Comment (Keith-CY) [inline website/docs/02-tx-skeleton.md]:
```suggestion
// broadcast the transaction
```

File: website/docs/02-tx-skeleton.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+const txHash = await rpc.sendTransaction(signedTransaction)
```
Comment (Keith-CY) [inline website/docs/02-tx-skeleton.md:46]: `rpc` is missing in the code snippet.

File: website/docs/02-tx-skeleton.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        output1["lock: Bob\ntype: UDT\ndata: 1(u128)"]
+    end
+
+
+
+    input0 -.-> output0
+    input0 -.-> output1
+```
+
+FOr a detailed explanation of CKB transaction structure, refer to [RFC-0022](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md).
```
Comment (Keith-CY) [inline website/docs/02-tx-skeleton.md]:
```suggestion
For a detailed explanation of CKB transaction structure, refer to [RFC-0022](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md).
```

File: website/docs/02-tx-skeleton.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+  signingEntries: ImmutableList<{
+    // the "witness_args_lock" is used to work with WitnessArgs.lock in @ckb-lumos/common
+    type: string
+    index: number
+    message: string
+  }>
+}
+```
+
+**Creating a TransactionSkeleton**:
```
Comment (Keith-CY) [inline website/docs/02-tx-skeleton.md]:
```suggestion
**Create a TransactionSkeleton**:
```

File: website/docs/02-tx-skeleton.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+- `headerDeps`: Include block header dependencies (e.g., CKB DAO deposit block).
+
+### `witnesses`
+
+One-time state for the transaction, such as signatures or preimages of hashes in `data`.
+
+### `signingEntries` and `fixedEntries`
+
+- `signingEntries`: An extension for handling transaction signing
+- `fixedEntries`: Cells that have been tagged with `fixedEntries`, including previous cells, will no longer participate in later calculations, such as `payFee`, but will only use cells after the `fixedEntries`.
```
Comment (Keith-CY) [inline website/docs/02-tx-skeleton.md]: I didn't get it.

File: website/docs/02-tx-skeleton.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
+---
+title: Transaction Skeleton
+sidebar_position: 3
+---
+
+## Transaction Skeleton
+
+Lumos provided an opinionated transaction structure called `TransactionSkeleton` from the `@ckb-lumos/helpers` module.
```
Comment (twhy) [inline website/docs/02-tx-skeleton.md]: Lumos provides

**Comment by homura** [comment]
@Keith-CY I fixed the typos and added an example to explain `fixedEntries`. Please check it again

**Review by Keith-CY — APPROVED** [review]

**Review by twhy — APPROVED** [review]

File: website/docs/02-tx-skeleton.md (hunk 7 above)
Comment (homura) [reply website/docs/02-tx-skeleton.md]: done

**Pull request merged**